lte/gateway/c/session_manager/PipelinedClient.cpp

```diff
@@ -41,7 +43,47 @@ magma::SessionSet create_session_set_req(
   req.mutable_node_id()->set_node_id_type(magma::NodeID::IPv4);
   req.mutable_state()->set_state(info.state);
 
-  for (const auto& final_req : info.pdr_rules) {
+  for (auto& final_req : info.pdr_rules) {
+    for (const auto& val : rules_to_activate) {
+      magma::VersionedPolicy versioned_rule;
+      versioned_rule.set_version(val.version);
+      versioned_rule.mutable_rule()->set_id(val.rule.id());
+      versioned_rule.mutable_rule()->set_priority(val.rule.priority());
+      for (const auto& flow : val.rule.flow_list()) {
+        if (final_req.pdi().net_instance().find("uplink") !=
+            std::string::npos) {
+          if (flow.match().direction() ==
+              magma::lte::FlowMatch_Direction_UPLINK) {
+            versioned_rule.mutable_rule()->mutable_flow_list()->Add()->CopyFrom(
+                flow);
+          }
+        } else {
+          if (flow.match().direction() ==
+              magma::lte::FlowMatch_Direction_DOWNLINK) {
+            versioned_rule.mutable_rule()->mutable_flow_list()->Add()->CopyFrom(
+                flow);
+          }
+        }
+      }
+      versioned_rule.mutable_rule()->mutable_qos()->CopyFrom(val.rule.qos());
+      versioned_rule.mutable_rule()->set_tracking_type(
+          val.rule.tracking_type());
+      // versioned_rule.mutable_rule()->CopyFrom(val.rule);
+      final_req.mutable_activate_flow_req()
+          ->mutable_policies()
+          ->Add()
+          ->CopyFrom(versioned_rule);
+    }
+    for (const magma::RuleToProcess& val : rules_to_deactivate) {
+      magma::VersionedPolicyID versioned_policy;
+      versioned_policy.set_version(val.version);
+      versioned_policy.set_rule_id(val.rule.id());
+      final_req.mutable_deactivate_flow_req()
+          ->mutable_policies()
+          ->Add()
+          ->CopyFrom(versioned_policy);
+    }
+
```

Comment:
for my own education, can you please tell me how this `set` function works related with pdr_rules?

Are `info.pdr_rules` the rules currently installed and `rules_to_activate` the ones to be activated?

Reply:
what is the relationship between `info.pdr_rules` and `rules_to_activate`. I am trying to understand the nested loop

Reply:
@prabinakpattnaik is that ^^^ your answer? Is it possible you edited my comment by mistake?

Reply:
The question I still have is `info.pdr_rules` the rules that current exist on the system and `rules_to_activate` the rules that need to be activated per this request

Reply:
Ohh..Sorry for that.

Yes, for each pdr is having rules_to_activate and that is activated the rules as per request.

info.pdr_rules is having two pdrs (One is uplink and another is downlink).